Bug: DOCUMENT_FROM_FILE can't be used with the new [[SECTION]] syntax

## Describe the bug

When including a document within a section using the new [[SECTION]] syntax, the compilation fails. With the old [SECTION] syntax, I get a deprecation warning, but the document compiles fine.
```
raise AssertionError(including_document_or_section)
AssertionError: SDocCompositeNode(node_type = "SECTION", fields = [1 elements], relations = [], section_contents = [1 elements], node_type_close = "SECTION", _tx_position = 57, _tx_position_end = 154, parent = SDocCompositeNode(...), is_composite = True, has_meta = False, fields_as_parsed = [1 elements], ordered_fields_lookup = {1 elements}, ng_level = 2, ng_document_reference = DocumentReference(...), ng_including_document_reference = DocumentReference(...), ng_line_start = 7, ng_line_end = 14, ng_col_start = 1, ng_col_end = 1, ng_byte_start = 57, ng_byte_end = 154, context = SDocNodeContext(...), reserved_mid = "056dadd2b4d3420c99d809edd4546241", mid_permanent = False, ng_resolved_custom_level = NoneType(...), custom_level = NoneType(...), ng_whitelisted = True, ng_has_requirements = False)
```

## To Reproduce

Using the old [SECTION] syntax, this kind of compilation succeeds just fine.
```
[DOCUMENT]
TITLE: StrictDoc

[SECTION]
TITLE: Introduction

  [SECTION]
  TITLE: System requirements
  
    [DOCUMENT_FROM_FILE]
    FILE: system_requirements.sdoc
    
  [/SECTION]
  
[/SECTION]
  ```

Using the new [[SECTION]] syntax, this kind of compilation fails with an assertion error.
```
[DOCUMENT]
TITLE: StrictDoc

[[SECTION]]
TITLE: Chapter
  
  [[SECTION]]
  TITLE: Subchapter
  
    [DOCUMENT_FROM_FILE]
    FILE: system_requirements.sdoc
  
  [[/SECTION]]

[[/SECTION]]
```

The file I am including in the previous two documents.
```
[DOCUMENT]
TITLE: System requirements

[GRAMMAR]
ELEMENTS:
- TAG: REQUIREMENT
  FIELDS:
  - TITLE: UID
    TYPE: String
    REQUIRED: True
  - TITLE: STATUS
    TYPE: SingleChoice(Draft, Active, Deleted)
    REQUIRED: True
  - TITLE: STATEMENT
    TYPE: String
    REQUIRED: True
  - TITLE: VERIFICATION
    TYPE: MultipleChoice(Review, Analysis, Inspection, Test)
    REQUIRED: True
  - TITLE: RATIONALE
    TYPE: String
    REQUIRED: True
  RELATIONS:
  - TYPE: Parent
  - TYPE: Child

[REQUIREMENT]
UID: TEST-SYS-001
STATUS: Draft
STATEMENT: The system shall have a requirement.
VERIFICATION: Analysis, Test
RATIONALE: Lorem Ipsum.
```

## Expected behaviour
I would expect the new [[SECTION]] to behave as the old [[SECTION]] syntax. If it's intended that I can't include documents within sections. How could I reproduce the old behaviour using the new syntax?

## Additional info
strictdoc version: 0.9.4

--- a/strictdoc/backend/sdoc/models/document_from_file.py
+++ b/strictdoc/backend/sdoc/models/document_from_file.py
@@ -13,6 +13,7 @@ from strictdoc.backend.sdoc.models.model import (
     SDocNodeIF,
     SDocSectionIF,
 )
+from strictdoc.backend.sdoc.models.node import SDocCompositeNode
 from strictdoc.helpers.auto_described import auto_described
 
 
@@ -80,7 +81,9 @@ class DocumentFromFile(SDocDocumentFromFileIF):
 
         if isinstance(including_document_or_section, SDocDocumentIF):
             including_document = including_document_or_section
-        elif isinstance(including_document_or_section, SDocSectionIF):
+        elif isinstance(
+            including_document_or_section, (SDocSectionIF, SDocCompositeNode)
+        ):
             including_document_: Optional[SDocDocumentIF] = (
                 including_document_or_section.get_document()
             )

--- a/tests/integration/features/document_fragments/40_DOCUMENT_FROM_FILE_in_SECTION/index.sdoc
+++ b/tests/integration/features/document_fragments/40_DOCUMENT_FROM_FILE_in_SECTION/index.sdoc
@@ -1,0 +1,15 @@
+[DOCUMENT]
+TITLE: Hello world
+
+[[SECTION]]
+TITLE: Chapter
+
+[[SECTION]]
+TITLE: Subchapter
+
+[DOCUMENT_FROM_FILE]
+FILE: system_requirements.sdoc
+
+[[/SECTION]]
+
+[[/SECTION]]

--- a/tests/integration/features/document_fragments/40_DOCUMENT_FROM_FILE_in_SECTION/system_requirements.sdoc
+++ b/tests/integration/features/document_fragments/40_DOCUMENT_FROM_FILE_in_SECTION/system_requirements.sdoc
@@ -1,0 +1,32 @@
+[DOCUMENT]
+TITLE: System requirements
+
+[GRAMMAR]
+ELEMENTS:
+- TAG: REQUIREMENT
+  FIELDS:
+  - TITLE: UID
+    TYPE: String
+    REQUIRED: True
+  - TITLE: STATUS
+    TYPE: SingleChoice(Draft, Active, Deleted)
+    REQUIRED: True
+  - TITLE: STATEMENT
+    TYPE: String
+    REQUIRED: True
+  - TITLE: VERIFICATION
+    TYPE: MultipleChoice(Review, Analysis, Inspection, Test)
+    REQUIRED: True
+  - TITLE: RATIONALE
+    TYPE: String
+    REQUIRED: True
+  RELATIONS:
+  - TYPE: Parent
+  - TYPE: Child
+
+[REQUIREMENT]
+UID: TEST-SYS-001
+STATUS: Draft
+STATEMENT: The system shall have a requirement.
+VERIFICATION: Analysis, Test
+RATIONALE: Lorem Ipsum.

--- a/tests/integration/features/document_fragments/40_DOCUMENT_FROM_FILE_in_SECTION/test.itest
+++ b/tests/integration/features/document_fragments/40_DOCUMENT_FROM_FILE_in_SECTION/test.itest
@@ -1,0 +1,9 @@
+RUN: %strictdoc export %S --output-dir %T | filecheck %s --dump-input=fail
+CHECK: Published: Hello world
+
+RUN: %cat %T/html/%THIS_TEST_FOLDER/index.html | filecheck %s --dump-input=fail --check-prefix CHECK-HTML
+
+CHECK-HTML: Hello world
+CHECK-HTML: Chapter
+CHECK-HTML: Subchapter
+CHECK-HTML: The system shall have a requirement.

